test/tls-provider.c

```diff
@@ -783,7 +783,6 @@ static void *xor_dup(const void *vfromkey, int selection)
         xor_freekey(tokey);
         tokey = NULL;
     }
-    return tokey;
```

Comment:
How is this `return` statement redundant? It is the only `return` statement in the function, and the function is not `void`, it is defined as returning a `void *`.